experimental/instrumentation-xhr/src/instrumentation.ts

```diff
@@ -62,6 +63,15 @@ export class XHRInstrumentation extends BaseInstrumentation {
           return instrumentation.originalSend.apply(this, [body === undefined ? null : body]);
         }
 
+        // add Faro RUM header to the request headers
+        const sessionId = faro.api.getSession()?.id;
+        if (sessionId !== null) {
+          instrumentation.originalSetRequestHeader.apply(this, [
+            faroRumHeader,
+            makeFaroRumHeaderValue(sessionId as string),
```

Comment:
Why don't need to cast `sessionId` to a string.

I think you saw an error before because  you checked for strict null instead for null and undefined so TS could not derive from the condition if session id is undefined or string.